### DISCUSSION TITLE
Page 11 layout and use

### DISCUSSION THREAD

**Discussion by rebel7580** [HASwitchPlate/HASPone#132] [Q&A] [answered]
Hi -

The screen layouts (shown on the "old" GitHub pages) show the page 11 layout as having a large area defined as b4, and the usual buttons 1-3 at the bottom.

However, the latest hasp version has for page 11 a button b4 top left, and a slider b5 next to it, then a blank area and the usual b1-b3 at the bottom.  (although I haven't found documentation of the pages on the new Github.)

I was hoping to display some text in the large area, but it looks like it is not accessable. Assuming it would be "b6", writing text to it returns a 0x1a code.

Is it possible to access this area, and if so, how?

Thanks

**Comment by rebel7580** [comment]
After some browsing around (discord, etc.) I'm concluding that the change where the big area of page 11 is now not defined was due to memory needed elsewhere. But this is just a guess so far.

I was hoping to use this big space to put some detailed weather forecasts.

**Comment by aderusha** [comment, marked as answer]
You are 100% correct on that - the big empty space was left there to allow experimenting with manual object placement in a manner similar to the blueprints that place an icon on the screen.  The project is stone out of memory, the existing solution uses literally every byte of RAM and won't compile on current versions of the Nextion editor without freeing up yet even more RAM elsewhere in the project.

At some point I'll push out some blueprints that demonstrate how one can use p11, if you want to poke around at things check out the `xstr` command in the [Nextion Instruction Set reference](https://nextion.tech/instruction-set/).

**Reply by rebel7580** [reply]
OK, I'll try playing with xstr for that spot, since I'm already using that to do icons ala your blueprints.

**Reply by rebel7580** [reply]
FWIW, I've successfully gotten this to work as I wanted: to display weather forecasts. I send a set of 6 xstr messages, one for each line, adjusting the  x,y and text values accordingly. I use Button 4 to display an icon for that day's weather and,  when pressed, it selects which of two "pages" of forecast info to show. Slider 5 is used to select which day to show. Flicking it to the right (or pressing the right-hand side) cycles to the next day. Flicking left, the previous.

Here's a pic:
![20211023_225136](https://user-images.githubusercontent.com/30353207/141363439-9e3c9323-53e1-4db2-ba32-6080b00d0bf7.jpg)

You can also see the motion detector at the bottom right that turns on the display when motion is detected.

**Comment by rebel7580** [comment]
FWIW, here's the TCL code I used to get the above display. Simplified TCL (I left out some details), but you should be able to see what the code is doing w.r.t the **xstr** command without knowing TCL.
```
    set font 6
    # Set coordinates for the "blank space" on Page 11: 6 lines
    set xstart 0
    set xend 240
    set ystart 60
    set yend 205
    set yoff 30

    # create a list, one entry per display line.
    append text [clock format [expr {[clock seconds]+(86400*([string index $day end]-1))}] -format {%b %e}]
    lappend text [dict get $humtemp $top $day lc]
    lappend text "High: [dict get $humtemp $top $day hi], Low: [dict get $humtemp $top $day lo]"
    lassign [uviTextColor [dict get $humtemp $top $day uvi]] uvi
    lappend text "UVI: $uvi"
    lappend text "Cloud Cover: [dict get $humtemp $top $day clouds]%"
    lappend text "Prob. of Precip: [dict get $humtemp $top $day pop]%"
    
    # Move through each line
    for {set linenum 1} {$linenum < 7} {incr linenum} {
        set json [format {["delay=1","xstr %d,%d,%d,%d,%d,%d,%d,1,0,1,\"%s\""]} \
        $xstart [expr {$ystart + ($linenum - 1)*$yoff}] $xend  \
        [expr {$yend - ($linenum - 1)*$yoff}] \
        $font $pco $bco  \
        [lindex $text [expr {$linenum - 1}]]]

        mqttComm pub hasp/$display/command/json $json
    }
    
    set icon [dict get $IconNames $iconname icon]
    # clear icon in area b4
    set json [format {["delay=1","xstr 0,0,65,65,8,%d,%d,0,0,1,\"%s\""]} \
        $bco $bco ""]
    mqttComm pub hasp/$display/command/json $json
    # write new icon
    set json [format {["delay=1","xstr 0,0,65,65,8,%d,%d,1,1,3,\"%s\""]} \
        $pco $bco [encoding convertto utf-8 $icon]]
    mqttComm pub hasp/$display/command/json $json
}

```
